UefiCpuPkg/PiSmmCpuDxeSmm/PiSmmCpuStandaloneMm.c

```diff
@@ -9,6 +9,61 @@ SPDX-License-Identifier: BSD-2-Clause-Patent
 
 #include "PiSmmCpuCommon.h"
 
+//
+// RemainingTasks Done flag
+//
+BOOLEAN  mRemainingTasksDone = FALSE;
+
+/**
+  Perform the remaining tasks.
+
+**/
+VOID
+PerformRemainingTasks (
+  VOID
+  )
+{
+  EDKII_PI_SMM_MEMORY_ATTRIBUTES_TABLE  *MemoryAttributesTable;
+
+  if (!mRemainingTasksDone) {
+    PERF_FUNCTION_BEGIN ();
+
+    //
+    // gEdkiiPiSmmMemoryAttributesTableGuid should have been published after SmmCore dispatched all MM drivers (MmDriverDispatchHandler).
+    //
+    SmmGetSystemConfigurationTable (&gEdkiiPiSmmMemoryAttributesTableGuid, (VOID **)&MemoryAttributesTable);
+
+    //
+    // Set critical region attribute in page table according to the MemoryAttributesTable
+    //
+    if (MemoryAttributesTable != NULL) {
+      SetMemMapAttributes (MemoryAttributesTable);
+    }
+
+    if (IsRestrictedMemoryAccess ()) {

```

Comment:
Can we remove this check?
If a platform wants to allow all DRAM access from SMM env, MmIpl should produce proper Resource HOBs to tell that all DRAM is accessible.

Reply:
we have aligned: the function will be cleared in the furture patch to create full static pagetbale by default.